src/openforms/utils/checks.py

```diff
@@ -81,3 +84,58 @@ def check_missing_init_files(app_configs, **kwargs):
         )
 
     return errors
+
+
+def get_field_line_number(
+    serializer_class: Type[Serializer], field_name: str
+) -> int | None:
+    # Get source lines and the starting line number of the serializer class
+    source_lines, start_line_number = inspect.getsourcelines(serializer_class)
+
+    # Find the line containing the field definition
+    for i, line in enumerate(source_lines):
+        if line.strip().startswith(f"{field_name} ="):
+            return start_line_number + i  # Compute the exact line number
+
+    return None  # Return None if field definition is not found
+
+
+@register
+def check_serializer_non_required_charfield_allow_blank_true(app_configs, **kwargs):
+    """
+    https://github.com/open-formulieren/open-forms/issues/4927
+
+    Check for serializers.CharFields that have required=False, but not `allow_blank=True`
+    to avoid errors occurring when empty strings are passed from the frontend.
+    """
+    errors = []
+    serializers = get_subclasses(Serializer)
+    for serializer in serializers:
+        # TODO not sure if this issue can also occur for fields defined in Meta.fields?
```

Comment:
I think it respects the model field and `blank=False` will result in `required=True`